src/pages/collab/index.tsx

```diff
@@ -179,6 +203,45 @@ const MatchRequestPage = () => {
     console.log("join session");
   };
 
+  const router = useRouter();
+  const {
+    data: userData,
+    isLoading: isLoadingUserData,
+    error: errorFetchingUser,
+  } = api.user.getCurrentUser.useQuery();
```

Comment:
For this why can't we use useSession? With that we also don't need to handle the loading
Will have to rmb to add the check after all the hooks tho cause it can lead rendering with more hooks if not careful
https://next-auth.js.org/getting-started/client#usesession